docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,288 @@ keywords:
 
-The second thing you need to consider is how to design the UI of the extension point so it supports a scenario where multiple extensions are being added without breaking the UI.
+### Requirements
 
-Finally, consider if there is any information from the current view that should be shared with the extensions added to the extension point. It could be information from the current view that could let the extending plugin prefill values or other data in the functionality being added via the extension.
+- **an extension point ID** _(string)_
+  - In case it's an extension point in core Grafana, it must start with `grafana/`
+  - In case it's inside a plugin, it must start with `plugin/<PLUGIN_ID>/`
+  - It must be unique
+- **an app plugin** - _currently only app plugins can create extension points_
 
-## Create an extension point
+:::note
+
+When designing the UI make sure the extension point supports a scenario where multiple extensions can be added without breaking the UI. Also consider if there is any information from the current view that should be shared with the extensions added to the extension point. It could be information from the current view that could let the extending plugin pre-fill values or other data in the extension's functionality.
 
-Use the `getPluginLinkExtensions` method in `@grafana/runtime` to create an extension point within your plugin.
+:::
+
+## Create an extension point
 
 :::danger
 
 When you create an extension point in a plugin, you create a public interface for other plugins to interact with. Changes to the extension point ID or its context could break any plugin that attempts to register a link inside your plugin.
 
 :::
 
-The `getPluginLinkExtensions` method takes an object consisting of the `extensionPointId`, which must begin `plugins/<PLUGIN_ID>`, and any contextual information that you want to provide. The `getPluginLinkExtensions` method returns a list of extension links that your program can then loop over.
+You can easily create an extension point using the following functions (they live in `@grafana/runtime`) to fetch extensions for a certain extension point ID:
+
+### The `usePluginExtensions()` hook
+
+:::note
+
+In case the `context` object is created dynamically, make sure to wrap it into a `useMemo()` to prevent unnecesssary rerenders. [More info](#)
+
+:::
+
+The `usePluginExtensions` React hooks returns the list of extensions that are registered for a certain extension point ID. The hook dynamically updates its return value when the list of extensions changes. This behavior usually happens when extensions are registered during runtime due to dynamic plugin loading.
+
+#### Syntax
+
+```tsx
+usePluginExtensions(options);
+usePluginLinkExtensions(options); // Only returns extensions that have type `type="link"`
+usePluginComponentExtensions(options); // Only returns extensions that have type `type="component"`
```

Comment:
I personally find it odd that these are all under "usePluginExtensions()" subsection but I may be in the minority.

Reply:
Yeah, maybe you are right. I only put them here for now as their syntax, arguments and return values are basically the same and I wanted to avoid too much duplication. On the other hand if we move towards a direction of having an API reference, then we will need to separate them, you are right.